actionview/lib/action_view/helpers/date_helper.rb

```diff
@@ -90,12 +90,17 @@ module DateHelper
       #   distance_of_time_in_words(from_time, from_time + 50.minutes, scope: 'datetime.distance_in_words.short') # => "an hour"
       #   distance_of_time_in_words(from_time, from_time + 3.hours, scope: 'datetime.distance_in_words.short')    # => "3 hours"
       def distance_of_time_in_words(from_time, to_time = 0, options = {})
+        if [from_time, to_time].any?(&:nil?)
+          raise ArgumentError, "given time must not be nil"
+        end
+
         options = {
           scope: :'datetime.distance_in_words'
         }.merge!(options)
 
         from_time = from_time.to_time if from_time.respond_to?(:to_time)
         to_time = to_time.to_time if to_time.respond_to?(:to_time)
+        from_time, to_time = Time.at(from_time), Time.at(to_time)
```

Comment:
By adding `Numeric#to_time` you could then make the method setup something like this:

``` ruby
def distance_of_time_in_words(from, to = 0, options = {})
  unless from.respond_to?(:to_time)
    raise ArgumentError, "from must be a time-like value"
  end

  unless to.respond_to?(:to_time)
    raise ArgumentError, "to must be a time-like value"
  end

  from_time, to_time = [from.to_time, to.to_time].sort
  distance_in_minutes = ((to_time - from_time)/60.0).round
  distance_in_seconds = (to_time - from_time).round
```